app/controllers/api/v1/messages_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    # Controller class that is responsible for handling message requests.
+    class MessagesController < ApplicationController
+      before_action :authenticate_user!
+      before_action :set_message, except: :create
+      before_action :new, only: :create
+      before_action :authorize_message!
+      after_action :verify_authorized
+
+      def new
+        @message = Message.new(message_params)
+      end
+
+      # POST /messages or /messages.json
+      def create
+        if message_params.blank?
+          render json: { error: 'There was no message data passed in so your message could not be saved.' }, status: :unprocessable_entity
+        elsif @message.save
+          render json: @message, status: :created
+        else
+          render json: @message.errors, status: :unprocessable_entity
+        end
+      end
+
+      # PUT /messages/:id or /messages/:id.json
+      def update
+        if message_params.blank?
+          render json: { error: 'There was no message data passed in. Your message could not be saved.' }, status: :unprocessable_entity
+        elsif @message.update(message_params)
+          render json: true, status: :ok
+        else
+          render json: @message.errors, status: :unprocessable_entity
+        end
+      end
+
+      # DELETE /messages/:id or /messages/:id.json
+      def destroy
+        if @message.destroy
+          render json: true, status: :no_content
+        else
+          render json: @message.errors, status: :unprocessable_entity
+        end
+      end
+
+      private
+
+      def set_message
+        @message = Message.find(params[:id])
+      end
+
+      def message_params
+        params.permit(:text, :conversation_id, :user_id)
```

Comment:
missing require

Reply:
permitting `:user_id` allows us to send messages from another person.
we should take user_id from current_user